src/hotspot/share/utilities/powerOfTwo.hpp

```diff
@@ -121,10 +121,17 @@ inline T next_power_of_2(T value)  {
 }
 
 // Find log2 value greater than this input
-template <typename T, ENABLE_IF(std::is_integral<T>::value)>
-inline T ceil_log2(T value) {
-  T ret;
-  for (ret = 1; ((T)1 << ret) < value; ++ret);
+template <typename T, typename U = typename std::make_unsigned<T>::type, ENABLE_IF(std::is_integral<T>::value)>
+inline int ceil_log2(T value) {
```

Comment:
Pre-existing: I think this should have been called `log2i_ceil`, with a description something like:
```
// Ceiling of log2 of a positive, integral value, i.e., smallest i such that value <= 2^i.
```
And move it near the other log2i variants.

This could be done as a followup, to keep this PR focused on the issue at hand.

Reply:
FYI, I filed [8344568](https://bugs.openjdk.org/browse/JDK-8344568) to track this.